presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkConfInitializer.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.SparkContext;
+
+public class PrestoSparkConfInitializer
+{
+    private static final String INITIALIZED_MARKER = "presto.spark.conf.initialized";
+
+    private PrestoSparkConfInitializer() {}
+
+    public static void initialize(SparkConf sparkConf)
+    {
+        if (sparkConf.get(INITIALIZED_MARKER, null) != null) {
```

Comment:
is this the standard trick in SparkConf? (set a marker )

Reply:
I don't know.

Let me elaborate on the problem I'm trying to solve.

The `SparkConf` must be initialized before the SparkContext is created. The `PrestoSparkRunner` accepts the `SparkContext` itself. The `SparkContext` must be pre-initialized. This check is needed to enforce that the `SparkConf` is properly initialized prior creating the `SparkContext`